src/cli/cli_udp.cpp

```diff
@@ -120,6 +188,61 @@ template <> otError UdpExample::Process<Cmd("open")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli udp send
+ * @code
+ * udp send hello
+ * Done
+ * @endcode
+ * @code
+ * udp send -t hello
+ * Done
+ * @endcode
+ * @code
+ * udp send -x 68656c6c6f
+ * Done
+ * @endcode
+ * @code
+ * udp send -s 800
+ * Done
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 hello
+ * Done
+ * @endcode
+ * @code
+ * udp send 172.17.0.1 1234 hello
+ * Sending to synthesized IPv6 address: fdde:ad00:beef:2:0:0:ac11:1
+ * Done
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -t hello
+ * Done
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -x 68656c6c6f
+ * Done
+ * @endcode
+ * @code
+ * udp send fdde:ad00:beef:0:bb1:ebd6:ad10:f33 1234 -s 800
+ * Done
+ * @endcode
+ * @cparam udp send [@ca{ip} @ca{port}] [@ca{type}] @ca{value}
```

Comment:
The way you have the parameters... does it mean that IP and Port must be specified together, but are also optional?  I.e. you wouldn't specify just the IP?

Can you make that clear in the description, that if you specify an IP you have to specify the Port.

Also, where does a message get sent if you don't specify an IP/port ... to all nodes in the Thread network?

Reply:
the message is sent using the socket. If `[ip port]` are not given the socket peer addresses are used (from `udp connect`).